biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeToProteinMapper.java

```diff
@@ -0,0 +1,193 @@
+        	}
+        	else {
+        		j++;
+        	}
+        }
+        
+        // remove untranslated regions from exons
+        int nExons = exonStarts.size();
+        exonStarts.remove(0);
+        exonStarts.add(0, cdsStart);
+        exonEnds.remove(nExons-1);
+        exonEnds.add(cdsEnd);
+    	
+        List<Range<Integer>> cdsRegion = new ArrayList<Range<Integer>>();
+        for ( int i=0; i<nExons; i++ ) {
+        	Range<Integer> r = Range.closed(exonStarts.get(i), exonEnds.get(i));
+        	cdsRegion.add(r);
+        }
+		return cdsRegion;
+    }
+    
+    /** Extracts the DNA sequence transcribed from the input genetic coordinates.
+    *
+    * @param gcp The container with chromosomal positions
+    * 
+    * @return the DNA sequence transcribed from the input genetic coordinates
+   */
+    public String getTranscriptSequence(GeneChromosomePosition gcp) throws IOException, CompoundNotFoundException {
+    	return getTranscriptSequence(gcp.getExonStarts(), gcp.getExonEnds(), gcp.getCdsStart(), gcp.getCdsEnd(), gcp.getOrientation());
+    }
+    
+    /** Extracts the DNA sequence transcribed from the input genetic coordinates.
+    *
+    * @param exonStarts The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)  
+    * @param exonEnds The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)
+    * @param cdsStart The start position of a coding region
+    * @param cdsEnd The end position of a coding region
+    * @param orientation The orientation of the strand where the gene is living
+    * 
+    * @return the DNA sequence transcribed from the input genetic coordinates
+    */
+	public String getTranscriptSequence(List<Integer> exonStarts, List<Integer> exonEnds, int codingStart, int codingEnd, Character orientation) throws IOException, CompoundNotFoundException {
+
+		List<Range<Integer>> cdsRegion;
+		if (orientation.equals("-")) {
+			cdsRegion = getCDSRegionsReverse(exonStarts, exonEnds, codingStart, codingEnd);
+		}
+		else {
+			cdsRegion = getCDSRegionsForward(exonStarts, exonEnds, codingStart, codingEnd);
+		}
+
+		String transcription = "";
+		for (Range<Integer> range : cdsRegion) {
+			int length = range.upperEndpoint() - range.lowerEndpoint();
+			transcription += parser.loadFragment(range.lowerEndpoint(), length);
+		}
+		if (orientation.equals("-")) {
+			transcription = new StringBuilder(transcription).reverse().toString();
+			DNASequence dna = new DNASequence(transcription);
+			SequenceView<NucleotideCompound> compliment = dna.getComplement();
+			transcription = compliment.getSequenceAsString();
+		}
+		return transcription;
+	}
+	
+    /** Converts the DNA sequence to protein sequence.
+    *
+    * @param sequence the DNA sequence
+    * 
+    * @return the protein sequence
+    */
+	public String convertDNAtoProteinSequence(String sequence) throws CompoundNotFoundException {
```

Comment:
name the argument dnaSequence to make it clear in javadoc that the input should be a DNA sequence.